src/server/string_family.cc

```diff
@@ -666,6 +683,66 @@ void StringFamily::GetSet(CmdArgList args, ConnectionContext* cntx) {
   return (*cntx)->SendNull();
 }
 
+void StringFamily::GetEx(CmdArgList args, ConnectionContext* cntx) {
+  string_view key = ArgS(args, 1);
+
+  DbSlice::ExpireParams exp_params;
+  int64_t int_arg = 0;
+
+  for (size_t i = 2; i < args.size(); i++) {
+    string_view cur_arg = ArgS(args, i);
+
+    i++;
+    if (i >= args.size()) {
+      return (*cntx)->SendError(kSyntaxErr);
+    }
+
+    string_view ex = ArgS(args, i);
+    if (!absl::SimpleAtoi(ex, &int_arg)) {
+      return (*cntx)->SendError(kInvalidIntErr);
+    }
+
+    if (int_arg <= 0) {
```

Comment:
looks much better!

question: why not implement the "persist" option? it's just a few more lines and you got the whole command covered.

Reply:
Now it may support `persist` option, checked by `TTL` command.

Reply:
> looks much better!
> 
> question: why not implement the "persist" option? it's just a few more lines and you got the whole command covered.



> looks much better!
> 
> question: why not implement the "persist" option? it's just a few more lines and you got the whole command covered.

Now it may support persist option, checked by TTL command.

Reply:
> looks much better!
> 
> question: why not implement the "persist" option? it's just a few more lines and you got the whole command covered.

Now it may support `persist` option, and the functionality was checked by `TTL` command.